server/service/src/requisition_line/response_requisition_line/insert/mod.rs

```diff
@@ -0,0 +1,279 @@
+                &context,
+                inline_init(|r: &mut InsertResponseRequisitionLine| {
+                    r.requisition_id = new_response_requisition().id;
+                    r.item_id = "invalid".to_string();
+                }),
+            ),
+            Err(ServiceError::ItemDoesNotExist)
+        );
+
+        // NotThisStoreRequisition
+        context.store_id = mock_store_b().id;
+        assert_eq!(
+            service.insert_response_requisition_line(
+                &context,
+                inline_init(|r: &mut InsertResponseRequisitionLine| {
+                    r.requisition_id = new_response_requisition().id;
+                }),
+            ),
+            Err(ServiceError::NotThisStoreRequisition)
+        );
+
+        // CannotAddItemToProgramRequisition
+        context.store_id = mock_store_a().id;
+        assert_eq!(
+            service.insert_response_requisition_line(
+                &context,
+                inline_init(|r: &mut InsertResponseRequisitionLine| {
+                    r.id = "some mock program line".to_string();
+                    r.requisition_id = program_requisition().id;
+                }),
+            ),
+            Err(ServiceError::CannotAddItemToProgramRequisition),
+        )
+    }
+
+    #[actix_rt::test]
+    async fn insert_response_requisition_line_success() {
+        let (_, connection, connection_manager, _) = setup_all_with_data(
+            "insert_response_requisition_line_success",
+            MockDataInserts::all(),
+            test_item_stats::mock_item_stats(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider
+            .context(mock_store_a().id, "".to_string())
+            .unwrap();
+        let service = service_provider.requisition_line_service;
+
+        service
+            .insert_response_requisition_line(
+                &context,
+                InsertResponseRequisitionLine {
+                    requisition_id: new_response_requisition().id,
+                    id: "new requisition line id".to_string(),
+                    item_id: test_item_stats::item2().id,
+                    their_stock_on_hand: Some(10.0),
+                    requested_quantity: Some(10.0),
+                    supply_quantity: Some(20.0),
+                    comment: Some("comment".to_string()),
+                },
+            )
+            .unwrap();
+
+        let line = RequisitionLineRowRepository::new(&connection)
+            .find_one_by_id("new requisition line id")
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(
+            line,
+            inline_edit(&line, |mut u| {
+                u.supply_quantity = 20.0;
+                u.requested_quantity = 10.0;
+                u.initial_stock_on_hand_units = 10.0;
+                u.available_stock_on_hand = test_item_stats::item_2_soh();
```

Comment:
Relating to my other comment about this field - this field is meant to be a snapshot of stock on hand of the customer when the requisition for stock was made. For an internal order it makes sense to look at item_stats as you have your stock data. When making a manual requisition for a facility or a store on another site (the latter being something rarely done), there is little to no chance knowing what their stock level is.

Long story short, relating to my other comment: this value should probably just be 10.0 to match the value in `initial_stock_on_hand`